packages/ember-runtime/lib/mixins/observable.js

```diff
@@ -306,6 +306,30 @@ export default Mixin.create({
     order and cause notifications to be delivered more often than you would
     like.
 
+    Property change notifications do not propogate over bindings if the object
+    returned is the same object (compared with ===). When a binding is created,
+    an observer is added on both the "from" and "to" paths. When
+    propertyDidChange is called on the "from" side, it triggers the "from"
+    observer which schedules a synchronization of the binding in the run loop
+    (into the sync queue). Synchronization just means to call set on the "to"
+    side of the binding with the value of the "from" side. This set call becomes
+    a no op if the value it is being set to is the same as it was previously.
```

Comment:
This is (essentially) the text that @mmun used to explain it to me, which was very helpful. I agree that the text certainly is helpful only to a user with a certain level of understanding.

The goal is to explain the relatively unintuitive behaviour that using `notifyPropertyChange` *does* fire observers and recompute CPs on the object the CP is on, but not on any objects that are bound to it. It seems to be something that comes up perennially, see #4487 #10661 #10405 and probably others.